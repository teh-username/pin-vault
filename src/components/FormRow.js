import React from 'react';
import { FormLabel, FormInput } from 'react-native-elements';

const FormRow = ({ label, name, value, onInputChange, editable = true }) => [
  <FormLabel key={`${name}_label`}>{label}</FormLabel>,
  <FormInput
    key={`${name}input`}
    value={value}
    onChangeText={data => onInputChange(name, data)}
    placeholder={`Please enter ${label.toLowerCase()}...`}
    editable={editable}
  />,
];

export default FormRow;
