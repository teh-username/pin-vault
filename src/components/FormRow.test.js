import React from 'react';
import { FormLabel, FormInput } from 'react-native-elements';

import FormRow from './FormRow';

describe('FormRow component test', () => {
    it('should render props correctly', () => {
        const mockCallback = jest.fn();

        const wrapper = shallow(
            <FormRow
                label="label"
                name="name"
                value="value"
                editable={false}
                onInputChange={mockCallback}
            />
        ).at(0);

        const rowLabel = wrapper.find(FormLabel);
        const rowInput = wrapper.find(FormInput);

        expect(rowLabel.prop('children')).toEqual('label');
        expect(rowLabel.key()).toEqual('name_label');

        expect(rowInput.key()).toEqual('name_input');
        expect(rowInput.prop('value')).toEqual('value');
        expect(rowInput.prop('placeholder')).toEqual('Please enter label...');
        expect(rowInput.prop('editable')).toEqual(false);

        rowInput.simulate('changeText', 'abcd');
        expect(mockCallback.mock.calls.length).toBe(1);
        expect(mockCallback.mock.calls[0][0]).toBe('name');
        expect(mockCallback.mock.calls[0][1]).toBe('abcd');

        expect(wrapper).toMatchSnapshot();
    });

    it('should default editable to true', () => {
        const wrapper = shallow(
            <FormRow label="label" name="name" value="value" />
        ).at(0);

        const rowInput = wrapper.find(FormInput);
        expect(rowInput.prop('editable')).toEqual(true);
    });
});
