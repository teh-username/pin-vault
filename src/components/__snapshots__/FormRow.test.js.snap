// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormRow component test should render props correctly 1`] = `
Array [
  <FormLabel
    key="name_label"
  />,
  <FormInput
    editable={false}
    key="name_input"
    normalizeFontSize={true}
    onChangeText={[Function]}
    placeholder="Please enter label..."
    value="value"
  />,
]
`;
